### DISCUSSION TITLE
Fixing DeprecationWarning:s in pyvsc

### DISCUSSION THREAD

**Discussion by jmcneal** [fvutils/pyvsc#227] [General]
I'm doing some very simple randomizations and I'm seeing DeprecationWarning: coming from several files in the VSC code.

Is there a better way to get these to go away without disabling all DeprecationWarnings?

I think I've got the latest code, I just DL'd it this week:
~projects/pyuvm/.venv/lib/python3.12/site-packages/pyvsc-0.9.3.10985030023.dist-info:

~/projects/pyuvm/.venv/lib/python3.12/site-packages/vsc/model/field_scalar_model.py:93: DeprecationWarning: __int__ returned non-int (type ValueInt).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  self.val.v = int(val)
~/projects/pyuvm/.venv/lib/python3.12/site-packages/vsc/model/variable_bound_in_propagator.py:31: DeprecationWarning: __int__ returned non-int (type ValueInt).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  in_r_l_t = list(map(lambda e:[int(e.lhs.val()),int(e.rhs.val())] if isinstance(e, ExprRangeModel)
~/projects/pyuvm/.venv/lib/python3.12/site-packages/vsc/model/expr_literal_model.py:42: DeprecationWarning: __int__ returned non-int (type ValueInt).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  return btor.Const(int(self.val()), ctx_width)
~/projects/pyuvm/.venv/lib/python3.12/site-packages/vsc/model/value_scalar.py:80: DeprecationWarning: __int__ returned non-int (type ValueInt).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  v = int(rhs)
~/projects/pyuvm/.venv/lib/python3.12/site-packages/vsc/model/expr_partselect_model.py:40: DeprecationWarning: __int__ returned non-int (type ValueInt).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  return btor.Slice(

**Comment by mballance** [comment]
Thanks for rains this, @jmcneal. I saw this pop up during testing (for some reason not during regular runs) recently, but hadn't had a chance to look into it.
PyVSC moved to wrapping primitive values (eg int) a while back. This allows it to support operations on those values. For example, you can slice the value returned from a VSC variable ('a' is a VSC variable. Python doesn't natively support slicing bits 7:2 of a value, but VSC does):
https://github.com/fvutils/pyvsc/blob/4476b835dbccf912c8c0f02b6c13520490891a1c/ve/unit/test_types.py#L307-L311

Doing a spot check of a couple of these, it appears that these are cases where an explicit int() conversion is requested on a 'wrapped' value:
https://github.com/fvutils/pyvsc/blob/4476b835dbccf912c8c0f02b6c13520490891a1c/src/vsc/model/expr_literal_model.py#L39-L42

In these cases, I suspect the int() explicit conversion can either be removed entirely, or should only be done selectively.
